### ISSUE TITLE
Hook_HealthRecalculation error on loading

### ISSUE THREAD

**Issue by AmaterasuOomikami** [PlexusDuMenton/RandomSpices#3]
Hello, hello.

An error comes up for me while loading RandomSpices at game launch. From what I can read, it looks like this:

```
[Info : BepinEx] 3/3 plugins to load
[Info : BepinEx] Loading [R2API 1.0]
[Info : Trace] Applied hook
[Info : BepinEx] Loading [ModRecalculate 0.4.0]
[Info : BepinEx] Loading [RandomSpices 0.1.4]
[Error : Unity Log] TypeLoadException: Could not resolve type with token 01000022 (from typeref, class/assembly Hook_HealthRecalculation, )
Stack trace:
UnityEngine.GameObject:AddComponent(Type)
BepInEx.Bootstrap.Chainloader:Start()
UnityEngine.Application.cctor()
RoR2.Console:RegisterLogHandler()
[Message: BepinEx] Chainloader startup complete
```
Perhaps relevant info:
- Health increases as it usually does, that is 60 health points per level as MUL-T: health recalculation thus does not seem to happen;
- [Pictures](https://imgur.com/a/qAZ1Dn0);
- This happens with a pirate copy of the game, windows 7, with the game executable located in User/Documents/Risk of Rain 2;
- BepInEx is up to date.

Wishing you a pleasant day, afternoon, evening, night,
Val-sama

**Comment by PlexusDuMenton** [comment]
Allready seen this bug one or two time on discord, but I honestly don't know what the cause since it only happen to some people, try reinstaling BepInEx or the game

**Comment by AmaterasuOomikami** [comment]
Hey hey.
Downloading and installing the game and BepInEx anew did not solve this issue. Thank you for your answer still, and good luck with further coding!
Val-sama

**Comment by PlexusDuMenton** [comment]
here some additional question : 
does using only "ModRecalculate" work ? 
Does other mod work too ?

**Comment by AmaterasuOomikami** [comment]
I had not checked before, but even RandomSpices' scaling of shield hitpoints does not work.

ModRecalculate alone triggers no error of any kind while loading, though I cannot check whether it is working since it has no direct effect on the game (as far as I understand).

I have yet to find a mod that does not work on my computer (other than this one, of course). So far I have installed InfiniteChance, StatsDisplay, AutoItemPickup, ScannerPlusOne, EmptyChestsBeGone, Multitudes, EnemyEdits and a bunch of other mods, and only the latter left me bewildered as to whether it was working as intended. No error of any kind appeared when loading these, either.

**Comment by PlexusDuMenton** [comment]
which version of ModRecalculate do you use ?

**Comment by AmaterasuOomikami** [comment]
I use the latest available version, always. (edit: and that for all software I use)
Checking earlier version: 0.2.0 that is labelled "preferred" for RandomSpices (RS) on thunderstore.io, along with RS' latest version, did not raise any error, made monsters extremely brittle (lemurians slain in one rebar puncher shot, which usually does not happen without items on rainstorm difficulty - I suppose their health was exponential as well), and made health scale as intended for my character (lv.1 health lower than vanilla and lv.3 onward health higher). That will teach me not to take advice into account.

Versions 0.3.0 triggered the error I was referring to in my opening post, wrecked health scaling, and removed my character's health regeneration. If the exponential health for monsters is intended as well, consider my problem solved as using version 0.2.0 does not cause any issue.

I should have tried a few more things before bothering you with this. My apologies.

**Comment by PlexusDuMenton** [comment]
np, monster exponential health is not realy intended. I'l ltake a look at it ^^
